src/main/java/org/simulator/io/CSVImporter.java

```diff
@@ -54,19 +48,63 @@
   private static final transient Logger logger = Logger.getLogger(CSVImporter.class.getName());
 
   /**
-   * The header of the file
+   * This method reads the data from the CSV file and converts it
+   * into the map with key as the column name and value as the array of amounts
+   * for the particular column.
+   * 
+   * @param pathname The path of the CSV file
+   * @return columnsMap -> LinkedListHashMap
    */
-  private String[] header;
+  private Map<String, double[]> readDataFromCSV(String pathname) throws IOException {
+    Map<String, double[]> columnsMap = new LinkedHashMap<>();
+    BufferedReader reader = new BufferedReader(new FileReader(pathname));
+    String line = reader.readLine();
+    List<String> lines = new LinkedList<String>();
+    String[] identifiers = line.split(",");
```

Comment:
I suggest to create another private method with one further argument, namely the separating character. We would call it with `','` (or possibly as a `String`?) assuming that commas are used as a separator. However, it gives us the flexibility to also use the method with other characters.